src/utils.mo

```diff
@@ -20,4 +23,20 @@ module {
       case (_) Prim.trap(message);
     };
   };
+
+  public func sliceIter<T>(iter : Iter.Iter<T>, limit : Nat, skip : Nat) : [T] {
+    var i = 0;
+    while (i < skip) {
+      let ?_ = iter.next() else return [];
+      i += 1;
+    };
+    let ret : Vec.Vector<T> = Vec.new();
+    i := 0;
+    label l while (i < limit) {
+      let ?x = iter.next() else break l;
+      Vec.add(ret, x);
+      i += 1;
+    };
+    Vec.toArray(ret);
```

Comment:
For this entire block, after we skipped, we can simply call `Iter.toArray(iter)`.

Reply:
fixed